src/C-DEngine/C-DCommunication/Protocols/WS/cde_WSCommon.cs

```diff
@@ -58,6 +58,7 @@ public void SetRequest(TheRequestData pRequest)
                 MySessionRequestData = pRequest;
             MySessionRequestData.WebSocket = this;
             mre = new ManualResetEventSlim(true);
+            mCancelToken = new();
```

Comment:
Should we cancel (and dispose) any prior canceltoken? 

Something like:
```cs
                var oldCancelToken = mCancelToken;
                mCancelToken = new();
                if (oldCancelToken != null)
                {
                    try
                    {
                        oldCancelToken.Cancel();
                    }
                    catch { }
                    try
                    {
                        oldCancelToken.Dispose();
                    }
                    catch { }
                }
```

Reply:
It’s not necessary. The whole problem was that the cancel token was null here in the first place. There is no way it is set there